src/main/java/org/commcare/core/graph/c3/DataConfiguration.java

```diff
@@ -373,15 +375,27 @@ private void setColumns(String xID, String yID, SeriesData s) throws GraphExcept
                 xValues.put(barIndex + 1);
                 mBarCount = Math.max(mBarCount, barIndex + 1);
                 if (addBarLabels) {
-                    mBarLabels.put(p.getX());
+                    if (!isRotatedBarGraph) {
+                        currentXLabelStr = p.getX();
+                    }
+                    mBarLabels.put(currentXLabelStr);
                 }
             } else {
                 if (mData.getType().equals(GraphUtil.TYPE_TIME)) {
-                    xValues.put(parseTime(p.getX(), description));
+                    currentXLabelStr = parseTime(p.getX(), description);
+                    xValues.put(currentXLabelStr);
                 } else {
-                    xValues.put(parseDouble(p.getX(), description));
+                    double val = parseDouble(p.getX(), description);
+                    xValues.put(val);
+                    currentXLabelStr = "" + val;

```

Comment:
nit: should use String.valueOf(), It's a bad practice to use String Concatenation for string conversion